go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -411,9 +418,27 @@ func (oc *DefaultNetworkController) addEgressFirewallRules(ef *egressFirewall, p
 			}
 		} else if len(rule.to.dnsName) > 0 {
 			// rule based on DNS NAME
-			dnsNameAddressSets, err := oc.egressFirewallDNS.Add(ef.namespace, rule.to.dnsName)
-			if err != nil {
-				return fmt.Errorf("error with EgressFirewallDNS - %v", err)
+			var dnsNameAddressSets addressset.AddressSet
+			var err error
+			// If DNSNameResolver is enabled, then use the resolver to get the address
+			// set corresponding to the DNS name, otherwise use the egressFirewallDNS
+			// to get the address set.
+			if config.OVNKubernetesFeature.EnableDNSNameResolver {
+				// Convert the DNS name to lower case fully qualified domain name.
+				dnsName := strings.ToLower(dns.Fqdn(rule.to.dnsName))
+				// Get the corresponding address set for the DNS name from the resolver
+				// without updating the current IPs added to the address set, if any.
+				dnsNameAddressSets, err = oc.resolver.Add(dnsName, nil, false)
```

Comment:
I would say this code should wait for an address set to be created by DNSNameResolver watcher instead of overwriting it from 2 different places. We could set a timeout for waiting and retry if an address set wasn't created on time.

Reply:
What should be the duration of the timeout?

Reply:
considering we don't have any perf-related data, we can just guess. Maybe 1 second for now?